sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -207,7 +207,7 @@ class StreamExecution(
       })
     } catch {
       case _: InterruptedException if state == TERMINATED => // interrupted by stop()
-      case NonFatal(e) =>
+      case e if NonFatal(e) || e.isInstanceOf[NoClassDefFoundError] =>
```

Comment:
Why only catch non-fatal (+ this one execption) here, given we are always going to record it and kill this thread?  Seems it would be better to always attempt to notify the user why we are dying?